integration/yarn/src/main/java/alluxio/yarn/Client.java

```diff
@@ -257,17 +258,50 @@ private void submitApplication() throws YarnException, IOException {
     monitorApplication();
   }
 
-  // Checks if the cluster has enough resource to launch application master
+  // Checks if the cluster has enough resource to launch application master,
+  // alluxio master and alluxio workers
   private void checkClusterResource(GetNewApplicationResponse appResponse) {
     int maxMem = appResponse.getMaximumResourceCapability().getMemory();
     int maxVCores = appResponse.getMaximumResourceCapability().getVirtualCores();
 
-    Preconditions.checkArgument(mAmMemoryInMB <= maxMem,
-        "ApplicationMaster memory specified above max threshold of cluster, specified="
-            + mAmMemoryInMB + ", max=" + maxMem);
-    Preconditions.checkArgument(mAmVCores <= maxVCores,
-        "ApplicationMaster virtual cores specified above max threshold of cluster, specified="
-            + mAmVCores + ", max=" + maxVCores);
+    if (mAmMemoryInMB <= maxMem) {
+      throw new RuntimeException(ExceptionMessage.YARN_NOT_ENOUGH_RESOURCES
+          .getMessage("ApplicationMaster", "memory", mAmMemoryInMB, maxMem));
+    }
+
+    if (mAmVCores <= maxVCores) {
+      throw new RuntimeException(ExceptionMessage.YARN_NOT_ENOUGH_RESOURCES
+          .getMessage("ApplicationMaster", "virtual cores", mAmVCores, maxVCores));
+    }
+
+    int masterVCores = Configuration.getInt(PropertyKey.INTEGRATION_MASTER_RESOURCE_CPU);
```

Comment:
nit: consider declaring these variables immediately before their use so that we have a bunch of (get config value, check config value) pairs. Alternately we could do groups of (get memory, get cores, check memory, check cores)